bionic/aip/future.py

```diff
@@ -0,0 +1,96 @@
+import blocks
```

Comment:
would you prefer to rely on the rest of bionic later on to not import this module unless we know we need it, or hide this potentially-error-raising imports inside the functions that require them? It moves the error from import time to runtime, but might help avoid some cases where it raises even though the functionality is not needed

Reply:
Good question about when we import this module. I think the simplest thing would probably be to import `blocks` with `import_optional_dependency` right before we use it, rather than taking care to import `aip` itself at the last minute. I don't think we can avoid waiting till runtime to surface the error, because we won't know whether the user intends to use AIP until they've created their flow.

Reply:
Exactly what Janek said. I would say that comment applies to all other optional imports in both `future` and `task` modules.

Reply:
I'll scope the imports to function level then, but definitely think we should follow up with some kind of instantiation time assertion that the dependencies are available to push this error up to as soon as we can know it will be an issue